### PR TITLE
chore(React InstantSearch): use InstantSearch + Insights integration in algolia-insights sample

### PR DESCRIPTION
**Pull request by tkrugg**
Other changes are:
- remove custom Hits component as no longer necessary
- update search-insights to v1.2.0
- update search-insights install snippets to extract source URL (as documented here: https://github.com/algolia/doc/pull/3063)

### CODE REVIEW

File: React InstantSearch/algolia-insights/src/App.js [hunk 1]
```diff
@@ -20,6 +20,8 @@ const searchClient = algoliasearch(
   'aadef574be1f9252bb48d4ea09b5cfe5'
 );
 
+const HitWithInsights = connectHitInsights(window.aa)(Hit);
```
Comment (Haroenv) [inline React InstantSearch/algolia-insights/src/App.js]: I'd move this closer to (probably just after) `Hit`, right now this order confused me

File: React InstantSearch/algolia-insights/src/App.js [hunk 2]
```diff
@@ -71,14 +59,8 @@ function Hit({ hit, index, searchResults }) {
       <button
         className="hit-action"
         onClick={() => {
-          window.aa('clickedObjectIDsAfterSearch', {
-            index: 'demo_ecommerce',
+          insights('clickedObjectIDsAfterSearch', {
             eventName: 'Add to favorite',
-            queryID: searchResults.queryID,
```
Comment (Haroenv) [inline React InstantSearch/algolia-insights/src/App.js]: <3 these changes!

File: React InstantSearch/algolia-insights/src/App.js (hunk 1 above)
Comment (samouss) [reply React InstantSearch/algolia-insights/src/App.js]: I agree @Haroenv, it's probably better to move the `Hit` inside its own file.

Comment (tkrugg) [reply React InstantSearch/algolia-insights/src/App.js]: done here: 41da299

**Pull request merged**